The upper restriction on the flit_core version appears to be a stock
generational restriction rather than a specific requirement, and the package
seems to build as expected anyway. Drop the requirement restriction.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["flit_core >=2,<3"]
+requires = ["flit_core"]
 build-backend = "flit_core.buildapi"
 
 [tool.flit.metadata]
